src/window.c

```diff
@@ -6314,6 +6334,104 @@ set_fraction(win_T *wp)
 				     + FRACTION_MULT / 2) / (long)wp->w_height;
 }
 
+/*
+ * Handle scroll position for 'nosplitscroll', replaces
+ * scroll_to_fraction() call from win_new_height().
+ * Instead we iterate over all windows in a tabpage and
+ * calculate the new scroll/cursor position.
+ * TODO: Ensure this also works with wrapped lines.
+ * Requires topline to be able to be set to a bufferline
+ * with some offset(row-wise scrolling/smoothscroll).
+ */
+    static void
+win_fix_scroll(win_T *next_curwin, int close)
+{
+    int      ls_offset = (p_ls == 1 && win_count() == 2) ? 1 : 0;
+    win_T    *wp;
+    linenr_T lnum;
+
+    FOR_ALL_WINDOWS(wp)
+    {
+	// Skip when window height has not changed or when
+	// buffer has less lines than the window height.
+	if (wp->w_height != wp->w_prev_height
+		&& wp->w_height < wp->w_buffer->b_ml.ml_line_count)
+	{
+	    lnum = wp->w_cursor.lnum;
+	    // Determine cursor position required to keep the same screenlines.
+	    if (wp->w_winrow != wp->w_prev_winrow)
+	    { // For bottom windows, keep the same botline except for 'ls' == 1.
+		if (!close && ((wp->w_winrow + VISIBLE_HEIGHT(wp)
+					+ wp->w_status_height) == cmdline_row))
```

Comment:
Is there any better way to determine whether a window is a bottom window i.e. above the cmdline? I would imagine something with vertical fr_layout and NULL fr_next but I'm not sure that's better and or correct. I guess a vertical frame can stop half-way the terminal window no? Then it wouldn't work, I haven't fully looked into the frame structure.

Reply:
Alternatively, it feels like there should be a more efficient way to determine whether this offset is possible in win_split_ins() such that we can get rid of this if/else clause. I see there is a need_status in win_split_ins() already. If we can use the pre-existing logic in win_split_ins() to add the need_status offset to w_prev_height, that might be enough. I haven't figured out the correct logic in win_split_ins() though, feels like it should be possible with flags/WSP_BOT/before variable and whatnot.